app/pages/project/affinity/AffinityPage.tsx

```diff
@@ -0,0 +1,162 @@
+  const { project } = getProjectSelector(params)
+  await queryClient
+    .fetchQuery(antiAffinityGroupList({ project }).optionsFn())
+    .then((data) => {
+      // Preload the anti-affinity group details
+      data.items.forEach((antiAffinityGroup) => {
+        queryClient.fetchQuery(
+          memberList({ antiAffinityGroup: antiAffinityGroup.name, project })
+        )
+      })
+    })
+  return null
+}
+
+const colHelper = createColumnHelper<AntiAffinityGroup>()
+
+export const AffinityPageHeader = ({ name = 'Affinity' }: { name?: string }) => (
+  <PageHeader>
+    <PageTitle icon={<Affinity24Icon />}>{name}</PageTitle>
+    {/* TODO: Add a DocsPopover with docLinks.affinity once the doc page exists */}
+  </PageHeader>
+)
+
+type AffinityGroupPolicyBadgeProps = { policy: AffinityPolicy; className?: string }
+const AffinityGroupPolicyBadge = ({ policy, className }: AffinityGroupPolicyBadgeProps) => (
+  <Badge
+    color="neutral"
+    variant={{ allow: 'default' as const, fail: 'solid' as const }[policy]}
+    className={className}
+  >
+    {policy}
+  </Badge>
+)
+
+export default function AffinityPage() {
+  const { project } = useProjectSelector()
+  const staticCols = [
+    colHelper.accessor('name', {
+      cell: makeLinkCell((antiAffinityGroup) =>
+        pb.antiAffinityGroup({ project, antiAffinityGroup })
+      ),
+    }),
+    colHelper.accessor('description', Columns.description),
+    colHelper.accessor(() => {}, {
+      header: 'type',
+      cell: () => <Badge>anti-affinity</Badge>,
+    }),
+    colHelper.accessor('policy', {
+      cell: (info) => <AffinityGroupPolicyBadge policy={info.getValue()} />,
+    }),
+    colHelper.accessor('name', {
+      header: 'members',
+      cell: (info) => <AffinityGroupMembersCell antiAffinityGroup={info.getValue()} />,
+    }),
+    colHelper.accessor('timeCreated', Columns.timeCreated),
+  ]
+  const { data } = usePrefetchedQuery(antiAffinityGroupList({ project }).optionsFn())
+
+  const table = useReactTable({
+    columns: staticCols,
+    data: data.items,
+    getCoreRowModel: getCoreRowModel(),
+  })
+
+  return (
+    <>
+      <AffinityPageHeader />
+      {data.items.length ? <Table table={table} /> : <AntiAffinityGroupEmptyState />}
+    </>
+  )
+}
+
+export const AntiAffinityGroupEmptyState = () => (
+  <TableEmptyBox>
+    <EmptyMessage
+      icon={<Affinity24Icon />}
+      title="No anti-affinity groups"
+      body="Create an anti-affinity group to see it here"
+      buttonText="New anti-affinity group"
+      buttonTo={pb.antiAffinityGroupNew(useProjectSelector())}
+    />
+  </TableEmptyBox>
+)
+
+// TODO: Use the prefetched query
+export const AffinityGroupMembersCell = ({
+  antiAffinityGroup,
+}: {
+  antiAffinityGroup: string
+}) => {
+  const { project } = useProjectSelector()
+  const { data: members } = useApiQuery('antiAffinityGroupMemberList', {
+    path: { antiAffinityGroup },
+    query: { project, limit: 2 },
+  })
```

Comment:
Check this out. You can still use the options helper with non-prefetched `useQuery`.

```suggestion
  const { data: members } = useQuery(memberList({ project, antiAffinityGroup }))
```

Reply:
Better to do it yourself — you need to add the import for `useQuery` anyway, and who knows whether what I wrote will typecheck or lint